src/main/scala/io/phdata/retirementage/loadgen/LoadGenerator.scala

```diff
@@ -14,11 +21,23 @@ object LoadGenerator {
 
   def generateTables(spark: SparkSession, conf: LoadGeneratorConfig): Unit = {
     // generate fact table
+    val factdf = generateTable(spark, conf.factCount.apply(), 1)
     // write fact table to disk
+    factdf.write
+      .mode(SaveMode.Overwrite)
+      .parquet(s"${conf.databaseName}.${conf.factName.apply()}")
     // generate dimension table
+    val dimdf = generateTableFromParent(spark, conf.dimensionCount.apply(), 3, factdf)
     // write dimension table to disk
+    dimdf.write
+      .mode(SaveMode.Overwrite)
+      .parquet(s"${conf.databaseName}.${conf.dimName.apply()}")
     // generate subdimension table
+    val sdf = generateTableFromParent(spark, conf.subDimensionCount.apply(), 1, dimdf)
     // write subdimension table to disk
+    sdf.write
```

Comment:
Use the longer `dimensionDf` and `subDimensionDf`  here, `dimdf` is arguably understandable, but `sdf` isn't.  The longer names can't hurt